docs/implement-check.md

```diff
@@ -0,0 +1,70 @@
+# Integrating Custom Check Into The Tool
+
+The Static Analysis Tool provides the necessary infrastructure for custom check implementations using the API of FindBugs, Checkstyle and PMD. 
+
+Helpful links when writing a custom check for the first time may be:
+
+- for [Checkstyle](http://checkstyle.sourceforge.net/writingchecks.html);
+- for [PMD](http://pmd.sourceforge.net/pmd-4.3.0/howtowritearule.html);
+- for [FindBugs](https://www.ibm.com/developerworks/library/j-findbug2/).
+
+In this guide we will use the Checkstyle API, because it is easy to use, supports different file extensions and languages. 
+
+We assume that you have already installed the Eclipse IDE and Maven.
+
+## Create an Eclipse project
+
+After checking out the repository on your local machine execute `mvn eclipse:eclipse` in project root directory, this will generate an Eclipse project, containing the `.classpath` and `.project` files. You are ready to import the project in Eclipse.
+
+```
+	Hint! Depending on your Java installation the static-code-analysis project might not find tools.jar. Please make sure that the folder containing the tools.jar is included in the PATH variable (for Windows) and rerun mvn eclipse:eclipse.
+```
+
+## Write Custom Checks Using Checkstyle API
+
+The first answer that you would have to answer before staring is what kind of files would you like to process:
+
+- for **none .java** files extend the `org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck`. We have included there some helpful methods for processing different types of files and others, take a look at the javadoc for detailed information;
+- for **.java** files you will most probably have to extend `com.puppycrawl.tools.checkstyle.api.AbstractCheck`.
+
+And once again, please refer to the Checkstyle documentation for writing a check, if you haven't.

```

Comment:
Add a link over the "Checkstyle documentation" to again redirect the contributor to the correct location and save time.